### ISSUE TITLE
Apollo may be produce incorrect data type for field with "optional boolean" type.

### ISSUE THREAD

**Issue by martintjandra** [apollographql/apollo-ios#455]
Hi. I use Apollo 0.9.4 and use Graphcool service to store my data. I have an object with field type `Boolean` in `types.graphql` which means it's an optional boolean.

```
type Check @model {
    ...
    isChecked: Boolean
    ...
}
```

Apollo generated this code in the `API.swift` for the corresponding getter method.

```
  public var isChecked: Swift.Optional<Bool?> {
    get {
      return graphQLMap["isChecked"] as! Swift.Optional<Bool?>
    }
    set {
      graphQLMap.updateValue(newValue, forKey: "isChecked")
    }
  }
```

I have confirmed that I can do mutation to create a data and the `isChecked` data is saved correctly as "true" / "false" in the graphcool.

The problem is when I tried to query the data and then access it, Xcode reported `signal SIGABRT` error on the `return graphQLMap["isChecked"] as! Swift.Optional<Bool?>` with detail: 

```
Could not cast value of type 'Swift.Optional<Swift.Bool>' (0x112994058) to 'Swift.Bool' (0x11296e1b8).
```

This is how I access the field in my code.

```
imageView.image = (checks[row].isChecked! ?? false) ? UIImage(named: "checked") : UIImage(named: "checkempty");
```

I still don't know if this also extends to another field with optional types. I think the correct type would be `Swift.Optional<Swift.Bool>`, and not `Swift.Optional<Swift.Bool?>` because this will produce double optionality.

**Comment by designatednerd** [comment]
Yeah the double optional is intentional - basically we're trying to determine "Was this not returned at all, or was this returned as `nil`?", since those are different cases in a graphQL response. We definitely shouldn't be using that `as!` in the graphql map though.

**Comment by designatednerd** [comment]
hello - [Tooling PR 1441](https://github.com/apollographql/apollo-tooling/pull/1441) removes the `as!` in getters for double-optionals. Please let me know if you have any suggestions further there!

**Comment by designatednerd** [comment]
Shipped with `0.14.0`!
